<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>947</width>
    <height>637</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="refresh_port_list">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset>
          <normaloff>Refresh.ico</normaloff>Refresh.ico</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="port_list"/>
      </item>
      <item>
       <widget class="QPushButton" name="open_comport">
        <property name="text">
         <string>Open</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="close_button">
        <property name="text">
         <string>Close</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QWidget" name="">
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Requests</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton">
            <property name="text">
             <string>Send Request</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Replies</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_2">
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <widget class="QTreeWidget" name="requests">
          <attribute name="headerCascadingSectionResizes">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Function</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Data</string>
           </property>
          </column>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QTreeWidget" name="replies">
          <column>
           <property name="text">
            <string>Type</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Data</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QPlainTextEdit" name="log">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>947</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>refresh_port_list</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>refresh_comports()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>21</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>57</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>open_comport</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>open_comport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>close_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close_comport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>send_request()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>403</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>7</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>replies</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>890</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>854</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refresh_comports()</slot>
  <slot>open_comport()</slot>
  <slot>close_comport()</slot>
  <slot>send_request()</slot>
 </slots>
</ui>
